### ISSUE TITLE
Use of Deprecated Constant Warning in Debian unstable 

### ISSUE THREAD

**Issue by NGC2023** [jcmgray/cotengrust#2]
Hi, While building in Debian unstable, I encountered the following warnings. pyo3 v0.22.2.
```
warning: use of deprecated constant `__pyfunction_ssa_to_linear::SIGNATURE`: this function has implicit defaults for the trailing `Option<T>` arguments 
         = note: these implicit defaults are being phased out 
         = help: add `#[pyo3(signature = (ssa_path, n=None))]` to this function to silence this warning and keep the current behavior
   --> src/lib.rs:868:4
    |
868 | fn ssa_to_linear(ssa_path: SSAPath, n: Option<usize>) -> SSAPath {
    |    ^^^^^^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default

warning: use of deprecated constant `__pyfunction_optimize_simplify::SIGNATURE`: this function has implicit defaults for the trailing `Option<T>` arguments 
         = note: these implicit defaults are being phased out 
         = help: add `#[pyo3(signature = (inputs, output, size_dict, use_ssa=None))]` to this function to silence this warning and keep the current behavior
   --> src/lib.rs:905:4
    |
905 | fn optimize_simplify(
    |    ^^^^^^^^^^^^^^^^^

warning: use of deprecated constant `__pyfunction_optimize_greedy::SIGNATURE`: this function has implicit defaults for the trailing `Option<T>` arguments 
         = note: these implicit defaults are being phased out 
         = help: add `#[pyo3(signature = (inputs, output, size_dict, costmod=None, temperature=None, seed=None, simplify=None, use_ssa=None))]` to this function to silence this warning and keep the current behavior
   --> src/lib.rs:922:4
    |
922 | fn optimize_greedy(
    |    ^^^^^^^^^^^^^^^

warning: use of deprecated constant `__pyfunction_optimize_random_greedy_track_flops::SIGNATURE`: this function has implicit defaults for the trailing `Option<T>` arguments 
         = note: these implicit defaults are being phased out 
         = help: add `#[pyo3(signature = (inputs, output, size_dict, ntrials, costmod=None, temperature=None, seed=None, simplify=None, use_ssa=None))]` to this function to silence this warning and keep the current behavior
   --> src/lib.rs:953:4
    |
953 | fn optimize_random_greedy_track_flops(
    |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated constant `__pyfunction_optimize_optimal::SIGNATURE`: this function has implicit defaults for the trailing `Option<T>` arguments 
         = note: these implicit defaults are being phased out 
         = help: add `#[pyo3(signature = (inputs, output, size_dict, minimize=None, cost_cap=None, search_outer=None, simplify=None, use_ssa=None))]` to this function to silence this warning and keep the current behavior
    --> src/lib.rs:1038:4
     |
1038 | fn optimize_optimal(
     |    ^^^^^^^^^^^^^^^^

warning: `cotengrust` (lib) generated 5 warnings
```
I have created a patch to address the warning related to the deprecated constant,
```patch
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -864,7 +864,7 @@ impl ContractionProcessor {
 
 // --------------------------- PYTHON FUNCTIONS ---------------------------- //
 
-#[pyfunction]
+#[pyfunction(signature = (ssa_path, n=None))]
 fn ssa_to_linear(ssa_path: SSAPath, n: Option<usize>) -> SSAPath {
     let n = match n {
         Some(n) => n,
@@ -901,7 +901,7 @@ fn find_subgraphs(
     cp.subgraphs()
 }
 
-#[pyfunction]
+#[pyfunction(signature = (inputs, output, size_dict, use_ssa=None))]
 fn optimize_simplify(
     inputs: Vec<Vec<char>>,
     output: Vec<char>,
@@ -918,7 +918,7 @@ fn optimize_simplify(
     }
 }
 
-#[pyfunction]
+#[pyfunction(signature = (inputs, output, size_dict, costmod=None, temperature=None, seed=None, simplify=None, use_ssa=None))]
 fn optimize_greedy(
     py: Python,
     inputs: Vec<Vec<char>>,
@@ -949,7 +949,7 @@ fn optimize_greedy(
     })
 }
 
-#[pyfunction]
+#[pyfunction(signature = (inputs, output, size_dict, ntrials, costmod=None, temperature=None, seed=None, simplify=None, use_ssa=None))]
 fn optimize_random_greedy_track_flops(
     py: Python,
     inputs: Vec<Vec<char>>,
@@ -1034,7 +1034,7 @@ fn optimize_random_greedy_track_flops(
     })
 }
 
-#[pyfunction]
+#[pyfunction(signature = (inputs, output, size_dict, minimize=None, cost_cap=None, search_outer=None, simplify=None, use_ssa=None))]
 fn optimize_optimal(
     py: Python,
     inputs: Vec<Vec<char>>,

```

**Comment by jcmgray** [comment]
Thanks the issue @NGC2023, should be fixed by https://github.com/jcmgray/cotengrust/commit/2627ac96e6da9d5e2023c6f14e805d7ab5f44fbc.
